includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -142,24 +142,27 @@ public function __construct(
 			add_action( 'woocommerce_checkout_subscription_created', [ $this, 'create_subscription' ] );
 			add_action( 'woocommerce_renewal_order_payment_complete', [ $this, 'create_subscription_for_manual_renewal' ] );
 			add_action( 'woocommerce_subscription_payment_method_updated', [ $this, 'maybe_create_subscription_from_update_payment_method' ], 10, 2 );
+
+			// Save the new token on the WCPay subscription when it's added to a WC subscription.
+			add_action( 'woocommerce_payment_token_added_to_order', [ $this, 'update_wcpay_subscription_payment_method' ], 10, 3 );
```

Comment:
I think this one needs to be moved to the `class_exists( 'WC_Subscription' )` check below.

If you have WC Subscriptions enabled and have some legacy Stripe Billing subscriptions, if you change the subscriptions payment method, it isn't updated in Stripe. 


Looking at the `update_wcpay_subscription_payment_method()` function, it looks safe to run on WC Subscription stores.

Reply:
Thanks @james-allan. Fixed in [b387737](https://github.com/Automattic/woocommerce-payments/pull/10358/commits/b387737663854e35ec8ca1b99c318bb45a007153)